go/pkg/yolosvc/driver_github.go

```diff
@@ -187,6 +195,48 @@ func (worker *githubWorker) fetchBaseObjects(ctx context.Context) (*yolopb.Batch
 	return batch, nil
 }
 
+func getOverrideBuildpb(owner string, repo string, artiID int64, token string) (*yolopb.OverrideBuild, error) {
+	override := &yolopb.OverrideBuild{}
+	// FIXME: is it always .zip files ?
+	req, err := http.NewRequest("GET", fmt.Sprintf("https://api.github.com/repos/%s/%s/actions/artifacts/%d/zip", owner, repo, artiID), nil)
+	if err != nil {
+		return nil, errors.New("getOverrideBuildpb: " + err.Error())
+	}
+
+	req.Header.Set("Accept", "application/vnd.github+json")
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", token))
+
+	do, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, errors.New("getOverrideBuildpb: " + err.Error())
+	}
+	body, err := io.ReadAll(do.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	zipReader, err := zip.NewReader(bytes.NewReader(body), int64(len(body)))
+	if err != nil {
+		return nil, errors.New("getOverrideBuildpb: " + err.Error())
+	}
```

Comment:
fyi, here is a good example of where you can stream res.Body directly in the zip.Reader.

Reply:
I see, understand the flow thanks to https://github.com/berty/yolo/blob/master/go/pkg/yolosvc/api_download.go#L265. I will try it in a more adapted contexte for sure. :+1: